snapcraft/internal/build_providers/_snap.py

```diff
@@ -316,13 +329,14 @@ def _record_revision(self, snap_name: str, snap_revision: str) -> None:
         else:
             self._registry_data[snap_name].append(entry)
 
-    def add(self, snap_name: str, snap_arch: str) -> None:
+    def add(self, snap_name: str, snap_arch: str, store_only: bool = False) -> None:
```

Comment:
Yeah, `inject_from_host: bool = True` feels like it makes more sense, here. However, do you ever envision creating a `SnapInjector` and calling this function while changing this parameter? I suspect not, which means perhaps this parameter should move to `__init__`. Honestly, that probably goes for `snap_arch` as well.

Reply:
certainly, we can move it back if needed.